test/Main.hs

```diff
@@ -308,3 +317,23 @@ voidstosTest =
            , datatypeCons    = []
            }
   )
+
+polyTest :: IO ()
+polyTest =
+  $(do info <- reifyDatatype 'MkPoly
+       let a = mkName "a"
+       validate info
+         DatatypeInfo
+           { datatypeName    = ''Poly
+           , datatypeContext = []
+           , datatypeVars    = [VarT a]
```

Comment:
You might find it unsettling that this is `VarT a` instead of, say, `SigT (VarT a) (VarT k)`. I do too, but this is an issue which affects data families on more versions of GHC than just 7.6/7.8, so I'll handle that separately.